lib/puppet/functions/dns_a.rb

```diff
@@ -16,14 +14,16 @@
   end
 
   def dns_a(record)
+    call_function('deprecation', 'dns_a', 'This method is deprecated please use the namspaced version dnsquery::a')
     Resolv::DNS.new.getresources(
```

Comment:
wonder if there is a way to just call dnsquery::a here instead of the duplicated code, possibly harder for the block case?

Reply:
https://www.puppet.com/docs/puppet/7/functions_ruby_implementation.html#calling-other-functions

Not tested:
```ruby
call_function('dnsquery::a', record)
```

Reply:
you would think i would have tried that considering the preceading line, Doh!.  anyway added and tests pass so seems to work.  thanks